### PR TITLE
Allow user/pass to be managed outside Puppet

### PR DESCRIPTION
**Pull request by Slashbunny**
The `value` field in `ini_setting` is optional. If it's undefined, it will ensure the setting is there, but not modify the value. The benefit to this type of behavior is that I can ensure there is a `password =` line
in the configuration, but allow the user to manually set it outside of Puppet. Future Puppet runs will not modify the user-entered value.

With the default being a blank string, Puppet will wipe out the user's changes every time Puppet runs.

Basically, this allows us to not have to deal with secrets such as the user's API key from within Puppet.

Another use case is if I am building an image for a developer to use- I may not know the username of the future developer, either, but I would still like to set everything else up so it's ready to use after filling in the username and password.

### CODE REVIEW

**Comment by ekohl** [comment]
The change makes sense to me but the tests are failing. Could you have a look?

**Comment by Slashbunny** [comment]
@ekohl Thanks, I corrected the failing tests.

**Comment by Slashbunny** [comment]
Tests are passing, so the label can be removed now.

File: manifests/init.pp [hunk 1]
```diff
@@ -53,8 +53,8 @@
 class jenkins_job_builder(
   $version                   = $jenkins_job_builder::params::version,
   Hash $jobs                 = $jenkins_job_builder::params::jobs,
-  String $user               = $jenkins_job_builder::params::user,
-  String $password           = $jenkins_job_builder::params::password,
+  Optional[String] $user     = $jenkins_job_builder::params::user,
```
Comment (bastelfreak) [inline manifests/init.pp]: Datatypes \o/

File: manifests/params.pp [hunk 2]
```diff
@@ -10,8 +10,8 @@
 class jenkins_job_builder::params {
 
   $jobs = {}
-  $user = ''
-  $password = ''
+  $user = undef
```
Comment (bastelfreak) [inline manifests/params.pp]: undef pattern \o/

**Comment by bastelfreak** [comment]
Thanks for this PR @Slashbunny !
